web/src/user/UserInterface.ts

```diff
@@ -364,4 +273,202 @@ export class UserInterface extends Interface {
+                <div class="pf-c-page__header-tools-group">
+                    <ak-action-button class="pf-m-warning pf-m-small" .apiRequest=${onClick}>
+                        ${msg("Stop impersonation")}
+                    </ak-action-button>
+                </div>
+            </div>`;
+    }
+}
+
+//  ___         _
+// | _ )_  _ __(_)_ _  ___ ______
+// | _ \ || (_-< | ' \/ -_|_-<_-<
+// |___/\_,_/__/_|_||_\___/__/__/
+//
+//
+@customElement("ak-interface-user")
+export class UserInterface extends Interface {
+    @state()
+    notificationDrawerOpen = getURLParam("notificationDrawerOpen", false);
+
+    @state()
+    apiDrawerOpen = getURLParam("apiDrawerOpen", false);
+
+    ws: WebsocketClient;
+
+    @state()
+    notificationsCount = 0;
+
+    @state()
+    me?: SessionUser;
+
+    constructor() {
+        super();
+        this.ws = new WebsocketClient();
+        this.fetchConfigurationDetails();
+        configureSentry(true);
+        this.toggleNotificationDrawer = this.toggleNotificationDrawer.bind(this);
+        this.toggleApiDrawer = this.toggleApiDrawer.bind(this);
+        this.fetchConfigurationDetails = this.fetchConfigurationDetails.bind(this);
+    }
+
+    connectedCallback() {
+        super.connectedCallback();
+        window.addEventListener(EVENT_NOTIFICATION_DRAWER_TOGGLE, this.toggleNotificationDrawer);
+        window.addEventListener(EVENT_API_DRAWER_TOGGLE, this.toggleApiDrawer);
+        window.addEventListener(EVENT_WS_MESSAGE, this.fetchConfigurationDetails);
+    }
+
+    disconnectedCallback() {
+        window.removeEventListener(EVENT_NOTIFICATION_DRAWER_TOGGLE, this.toggleNotificationDrawer);
+        window.removeEventListener(EVENT_API_DRAWER_TOGGLE, this.toggleApiDrawer);
+        window.removeEventListener(EVENT_WS_MESSAGE, this.fetchConfigurationDetails);
+        super.disconnectedCallback();
+    }
+
+    toggleNotificationDrawer() {
+        this.notificationDrawerOpen = !this.notificationDrawerOpen;
+        updateURLParams({
+            notificationDrawerOpen: this.notificationDrawerOpen,
+        });
+    }
+
+    toggleApiDrawer() {
+        this.apiDrawerOpen = !this.apiDrawerOpen;
+        updateURLParams({
+            apiDrawerOpen: this.apiDrawerOpen,
+        });
+    }
+
+    fetchConfigurationDetails() {
+        me().then((me: SessionUser) => {
+            this.me = me;
+            new EventsApi(DEFAULT_CONFIG)
+                .eventsNotificationsList({
+                    seen: false,
+                    ordering: "-created",
+                    pageSize: 1,
+                    user: this.me.user.pk,
+                })
+                .then((notifications) => {
+                    this.notificationsCount = notifications.pagination.count;
+                });
+        });
+    }
+
+    get isFullyConfigured() {
+        return !!(this.uiConfig && this.me);
+    }
+
+    render() {
+        if (!this.isFullyConfigured) {
+            return nothing;
+        }
+
```

Comment:
I like this because it makes it very clear what the presentation responds to: here are the state variables that will trigger a re-render.